### DISCUSSION TITLE
[>] Checking Python 3 installation....(None)

### DISCUSSION THREAD

**Discussion by CGMossa** [quarto-dev/quarto-cli#3846] [Q&A] [answered]
I've tried `quarto check` and it cannot detect my python installation. I find that odd since I have python installed. 
I've looked and there is nothing to configure or try otherwise.
I've tried

```powershell
$env:QUARTO_PYTHON = "python"
``` 

Doesn't work either. 

Here's the print of what I got:
```default
PS C:\Users\tpb398\Documents\GitHub\hexagon_paper> quarto check
[>] Checking Quarto installation......OK
      Version: 1.2.280
      Path: C:\Users\tpb398\scoop\apps\quarto\current\bin
      CodePage: 1252

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....(None)

      Unable to locate an installed version of Python 3.
      Install Python 3 from https://www.python.org/downloads/

[>] Checking R installation...........OK
      Version: 4.2.2
      Path: C:/Users/tpb398/scoop/apps/r/current
      LibPaths:
        - C:/Users/tpb398/scoop/persist/r/site-library
        - C:/Users/tpb398/scoop/apps/R/4.2.2/library
      rmarkdown: 2.19

[>] Checking Knitr engine render......OK

PS C:\Users\tpb398\Documents\GitHub\hexagon_paper> python
Python 3.10.5 (tags/v3.10.5:f377153, Jun  6 2022, 16:14:13) [MSC v.1929 64 bit (AMD64)] on win32
>>> exit()
```

**Comment by francisbarton** [comment]
I think the `QUARTO_PYTHON` variable needs to specify the path to the directory where your desired Python executable is installed

**Reply by mcanouil** [reply]
@francisbarton Thanks, but avoid replying to 3 years old zombie thread.

For reference:

- https://quarto.org/docs/advanced/environment-vars.html

**Reply by CGMossa** [reply]
Still alive though!

**Reply by cderv** [reply, marked as answer]
Thanks @francisbarton for your reply. Much appreciated to still help on unanswered topic! 

@CGMossa sorry we missed your question 3 years ago. If you still have issue nowadays, read the updated doc. As mention in installation (https://quarto.org/docs/computations/python.html#installation), quarto will try to be clever to detect Python from PATH, or from virtual activated, but you can still set env var to a python filepath (so not `$env:QUARTO_PYTHON = "python"` as it will look for `python` locally to your project). 

Running `quarto check jupyter --log-level debug` could give you hints on discovery, or `quarto render --log-level debug` (more in https://quarto.org/docs/troubleshooting/#verbose-mode) 

Hope it helps. And please to share again if you still have troubles setting up. 

Thank you both.
